### PR TITLE
DOC: add minimum dependency version policy in writing

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: doc/devel/min_dep_policy.rst [hunk 1]
```diff
@@ -0,0 +1,38 @@
+.. _min_deps_policy:
+
+======================================
+Minimum Version Of dependencies Policy
```
Comment (timhoffm) [inline doc/devel/min_dep_policy.rst]: Not quite sure, what the capitalization policy really is, but everything capitalized except "dependencies" is a bit weird.

[devdocs](https://matplotlib.org/devel/documenting_mpl.html#section-name-formatting) says

> For everything but top-level chapters, use Upper lower for section titles

Don't know if this is a top-level chapter (and if so, what the policy for these were.

File: doc/devel/min_dep_policy.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- support minor versions of ``Python`` initially released in the previous
+  3 years
+- support minor versions of ``numpy`` initially released in the
+  previous 3 years or oldest that supports the minimum python version
+  (which ever is higher)
+
+We will bump the minimum python and numpy versions as we can every
+minor and major release, but never on a patch release.
+
+Other Dependencies
```
Comment (timhoffm) [inline doc/devel/min_dep_policy.rst]: I would split this into separate sections "Python packages" and "System/c-dependencies".

File: doc/devel/min_dep_policy.rst (hunk 1 above)
Comment (tacaswell) [reply doc/devel/min_dep_policy.rst]: That is indeed a typo...

File: doc/devel/min_dep_policy.rst (hunk 2 above)
Comment (tacaswell) [reply doc/devel/min_dep_policy.rst]: I like that.

File: doc/devel/min_dep_policy.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+old versions no longer support our minimum numpy or python.
+
+
+System and C-dependencies
+=========================
+
+For system or c-dependencies (libpng, freetype, GUI frameworks, latex,
+gs, ffmpeg) support as old as practical.  These can be difficult to
+install for end-users and we want to be usable on as many systems as
+possible.  We will only bump these when absolutely necessary.
```
Comment (anntzer) [inline doc/devel/min_dep_policy.rst]: I would prefer a slightly less restrictive policy on C-level dependencies, something like "when bumping the dependency solves a real problem (for yet-to-be defined value of "real") and the new version is available on the latest (or whatever) Ubuntu LTS and Homebrew" (for example).

Comment (tacaswell) [reply doc/devel/min_dep_policy.rst]: Went with " We will bump these on a case-by-case basis." which is even vaguer.

**Comment by tacaswell** [comment]
This does not address the dependencies that we vendor, we can in principle be _more_ aggressive about bumping those as if we can not find a new enough version we provide it (but that would probably annoy the linux distributions).

**Pull request merged**